src/nodejs/jshttpclient.nim

```diff
@@ -1,5 +1,212 @@
-## HttpClient for JavaScript targets.
-## * Asyncronous `HttpClient` implemented on top of `jsfetch`.
-## * Syncronous `HttpClient` implemented on top of `jsxmlhttprequest`.
-import jsasynchttpclient, jssynchttpclient
-export jsasynchttpclient, jssynchttpclient
+## * HttpClient for JavaScript targets implemented as sugar on top of `fusion/js/jsxmlhttprequest <https://nim-lang.github.io/fusion/src/fusion/js/jsxmlhttprequest>` and `std/jsfetch <https://nim-lang.github.io/Nim/jsfetch>`_
+when not defined(js):
+  {.fatal: "Module jshttpclient is designed to be used with the JavaScript backend.".}
+
+import std/[asyncjs, jsheaders, jsfetch]
+import fusion/js/jsxmlhttprequest
+from std/uri import Uri
+
+type
+  JsHttpClient* = XMLHttpRequest
+  JsAsyncHttpClient* = ref object of JsRoot
+
+  JsRequest* = ref object of JsRoot
+    url*, `method`*, body*, integrity*, referrer*, mode*, credentials*, cache*, redirect*, referrerPolicy*: cstring
+    headers*: Headers
+    keepalive*: bool
+
+  JsResponse* = ref object of JsRoot
+    status*: cint
+    statusText*, url*, responseText*: cstring
+    headers*: Headers
+
+func newJsHttpClient*(): JsHttpClient {.importjs: "new XMLHttpRequest()".}
+
+func newJsAsyncHttpClient*(): JsAsyncHttpClient = discard
+
+func newJsRequest*(
+  url, `method`: cstring;
+  body, integrity: cstring = "";
+  referrer: cstring = "client";
+  mode: cstring = "cors";
+  credentials: cstring = "include";
+  cache: cstring = "default";
+  redirect: cstring = "follow";
+  referrerPolicy: cstring = "unsafe-url";
+  headers: Headers = newHeaders();
+  keepalive: bool = false
```

Comment:
This is too vertical, we have 4K monitors nowadays, no need for less than 80 char strict,
please make it more horizontal, wrap it like at ~100 chars or more.

Reply:
fixed